src/Nri/Ui/ClickableText/V4.elm

```diff
@@ -0,0 +1,721 @@
+    in
+    Nri.Ui.styled Html.button
+        (dataDescriptor "button")
+        (clickableTextSharedStyles config.disabled ++ clickableTextButtonStyles ++ config.customStyles)
+        (ClickableAttributes.toButtonAttributes config.clickableAttributes
+            { disabled = config.disabled }
+            ++ config.customAttributes
+        )
+        [ viewContent config Button ]
+
+
+{-| Creates a `<a>` element
+-}
+link :
+    String
+    -> List (Attribute msg)
+    -> Html msg
+link label_ attributes =
+    let
+        config =
+            (label label_ :: attributes)
+                |> List.foldl (\(Attribute attribute) l -> attribute l) defaults
+
+        ( name, clickableAttributes ) =
+            ClickableAttributes.toLinkAttributes
+                { routeToString = identity
+                , isDisabled = config.disabled
+                }
+                config.clickableAttributes
+    in
+    Nri.Ui.styled Html.a
+        (dataDescriptor name)
+        (clickableTextSharedStyles config.disabled ++ clickableTextLinkStyles ++ config.customStyles)
+        (clickableAttributes ++ config.customAttributes)
+        [ viewContent config Link ]
+
+
+viewContent :
+    { a
+        | label : String
+        , size : Size
+        , icon : Maybe Svg
+        , rightIcon : Maybe Svg
+        , iconStyles : List Style
+        , rightIconStyles : List Style
+    }
+    -> Kind
+    -> Html msg
+viewContent config kind =
+    let
+        fontSize =
+            case config.size of
+                Inherited ->
+                    Css.fontSize Css.inherit
+
+                Small ->
+                    Css.fontSize (Css.px 13)
+
+                Medium ->
+                    Css.fontSize (Css.px 15)
+
+                Large ->
+                    Css.fontSize (Css.px 18)
+
+        iconMarginRight =
+            case config.size of
+                Inherited ->
+                    Css.marginRight (Css.em 0.2)
+
+                Small ->
+                    Css.marginRight (Css.px 3)
+
+                Medium ->
+                    Css.marginRight (Css.px 3)
+
+                Large ->
+                    Css.marginRight (Css.px 4)
+
+        iconMarginLeft =
+            case config.size of
+                Inherited ->
+                    Css.marginLeft (Css.em 0.2)
+
+                Small ->
+                    Css.marginLeft (Css.px 3)
+
+                Medium ->
+                    Css.marginLeft (Css.px 3)
+
+                Large ->
+                    Css.marginLeft (Css.px 4)
+
+        viewIcon position icon_ =
+            icon_
+                |> Svg.withCss
+                    ([ Css.width (Css.em 1)
+                     , Css.height (Css.em 1)
+                     , Css.position Css.relative
+                     , Css.top (Css.em 0.1)
+                     ]
+                        ++ (case position of
+                                Left ->
+                                    iconMarginRight
```

Comment:
Given the position absolute style, I don't think that the margin right is doing anything when the clickable is a button